hocon/value.go

```diff
@@ -259,21 +272,28 @@ func (p *HoconValue) GetObject() (*HoconObject, error) {
 	}
 
 	if sub, ok := raw.(MightBeAHoconObject); ok {
-		if sub != nil && sub.IsObject() {
-			return sub.GetObject()
+		if sub != nil {
+			isObject, err := sub.IsObject()
+			if err != nil {
+				return nil, err
+			}
+
+			if isObject {
+				return sub.GetObject()
+			}
```

Comment:
i don't know but u use error interface not by truth way. In go specification exists special rule for this case. All function which return value and error should return or value, or error where if value not nil error should be nill, and anohter case where error not nil value should be not nil. By the way if u check original go sources u can find how can be using error interface and how u can handling this interface .  That's need rewriting by the another issue or maybe we can disquss of there in personal dialog